src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -201,38 +200,55 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
+        private async Task UpdateInstalledPackagesAsync() {
+            var token = await _buildIndexLock.WaitAsync();
```

Comment:
Yes, it tried to rebuild entire index but not because of the lock, but because `ScheduleIdleTimeRebuild` calls `RebuildIndexAsync`, which isn't called anymore. If you don't call `ResetAsync`, there can be concurrent calls to the `UpdateInstalledPackagesAsync` which may impact performance.
So I believe it should be 
```cs
var token = await _buildIndexLock.ResetAsync();
if (!token.IsSet) {
    try {
...
        token.Set();
    } catch (RHostDisconnectedException) {
    } finally {
        token.Reset();
    }
}
```